src/engine/engine_bind_test.cc

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <openssl/engine.h>
+#include <gtest/gtest.h>
+
+#include "src/engine/engine_bind.h"
+#include "src/engine/engine_name.h"
+
+namespace kmsengine {
+namespace engine {
+namespace {
+
+// Test fixture for instantiating an OpenSSL ENGINE object.
+class EngineBindTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    engine = ENGINE_new();
+    ASSERT_NE(engine, nullptr);  // Sanity check for malloc errors.
+  }
+
+  void TearDown() override {
+    ENGINE_free(engine);
+  }
+
+  ENGINE *engine;
+};
+
+TEST_F(EngineBindTest, SetsEngineId) {
```

Comment:
If all you do is engine_bind (i.e. the test body is not substantially different from each other), maybe just combine all the verifications into one?

Reply:
Do you mean something like this?

```c++
TEST_F(EngineBindTest, InitializesExpectedEngineStructFields) {
  EngineBind(engine, NULL);
  EXPECT_STREQ(ENGINE_get_id(engine), kEngineId);
  EXPECT_STREQ(ENGINE_get_name(engine), kEngineName);
  EXPECT_NE(ENGINE_get_init_function(engine), nullptr);
  EXPECT_NE(ENGINE_get_finish_function(engine), nullptr);
  EXPECT_TRUE(ENGINE_get_flags(engine) & ENGINE_FLAGS_NO_REGISTER_ALL);
}
```

Reply:
In this case, is it still worth it to have the text fixture that calls `ENGINE_new` and `ENGINE_free`?

Reply:
Yeah something like that. As for text fixture, that's up to you. Do you foresee other checks being added later on to this file that would require test fixtures?

Reply:
At the moment, I can't think of anything that can't just be added as another `EXPECT` so I'll just remove the fixture for now.